File: app/src/main/java/tpcreative/co/qrscanner/ui/seeyousoon/SeeYouSoonActivity.java
package tpcreative.co.qrscanner.ui.seeyousoon;
import android.os.Bundle;
import tpcreative.co.qrscanner.R;
import tpcreative.co.qrscanner.common.DelayShowUIListener;
import tpcreative.co.qrscanner.common.Utils;
import tpcreative.co.qrscanner.common.activity.BaseActivity;
import tpcreative.co.qrscanner.common.view.Bungee;

public class SeeYouSoonActivity extends BaseActivity {
    private final int DELAY_TO_SHOW_UI = 3000;
    private static final String TAG = SeeYouSoonActivity.class.getSimpleName();
    @Override
    protected void onCreate(Bundle savedInstanceState) {
        super.onCreate(savedInstanceState);
        setContentView(R.layout.activity_see_you_soon);
        Utils.onObserveVisitView(DELAY_TO_SHOW_UI, new DelayShowUIListener() {
            @Override
            public void onSetVisitView() {
                finish();
                Utils.Log(TAG,"See you soon");
            }
        });
    }

    @Override
    public void onBackPressed() {

    }
}

File: app/src/main/java/tpcreative/co/qrscanner/ui/seeyousoon/SeeYouSoonActivity.kt
package tpcreative.co.qrscanner.ui.seeyousoon
import android.os.Bundle
import kotlinx.coroutines.CoroutineScope
import kotlinx.coroutines.Dispatchers
import kotlinx.coroutines.delay
import kotlinx.coroutines.launch
import tpcreative.co.qrscanner.R
import tpcreative.co.qrscanner.common.Utils
import tpcreative.co.qrscanner.common.activity.BaseActivity

class SeeYouSoonActivity : BaseActivity() {
    private val DELAY_TO_SHOW_UI = 3000
    override fun onCreate(savedInstanceState: Bundle?) {
        super.onCreate(savedInstanceState)
        setContentView(R.layout.activity_see_you_soon)
        CoroutineScope(Dispatchers.Main).launch {
            delay(DELAY_TO_SHOW_UI.toLong())
            finish()
            Utils.Log(TAG, "See you soon")
        }
    }

    companion object {
        private val TAG = SeeYouSoonActivity::class.java.simpleName
    }
}